graph-proxy/src/graphql/workflows.rs

```diff
@@ -275,16 +275,60 @@ impl TryFrom<String> for TaskStatus {
     }
 }
 
+/// A task artifact
+#[allow(clippy::missing_docs_in_private_items)]
+#[derive(Debug)]
+struct Artifact<'a> {
+    manifest: &'a IoArgoprojWorkflowV1alpha1Artifact,
+    metadata: &'a Metadata,
+    node_id: &'a str,
+}
+
+#[Object]
+impl Artifact<'_> {
+    /// The name of the artifact
+    async fn name(&self) -> &str {
+        let path = std::path::Path::new(
+            self.manifest
+                .s3
+                .as_ref()
+                .expect("Artifacts are stored in an S3 bucket")
```

Comment:
Instead of panicking, could we return some error variant (e.g. `UnrecognisedArtifactStore`)?